src/client/Microsoft.Identity.Client/Internal/ClaimsHelper.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Identity.Json;
+using Microsoft.Identity.Json.Linq;
+
+namespace Microsoft.Identity.Client.Internal
+{
+    internal class ClaimsHelper
+    {
+        private const string AccessTokenClaim = "access_token";
+        private const string XmsClientCapability = "xms_cc";
+
+        internal static string MergeClaimsAndClientCapabilities(
+            string claims, 
+            IEnumerable<string> clientCapabilities)
+        {
+            if (clientCapabilities != null && clientCapabilities.Any())
+            {
+                JObject capabilitiesJson = CreateClientCapabilitiesRequestJson(clientCapabilities);
+                MergeClaimsAndCapabilityJson(claims, capabilitiesJson);
+
+                return capabilitiesJson.ToString(Formatting.None);
+            }
+
+            return claims;
+        }
+
+        private static void MergeClaimsAndCapabilityJson(string claims, JObject capabilitiesJson)
+        {
+            if (!string.IsNullOrEmpty(claims))
+            {
+                JObject claimsJson;
+                try
+                {
+                    claimsJson = JObject.Parse(claims);
+                }
+                catch (JsonReaderException ex)
+                {
+                    throw new MsalClientException(
+                        MsalError.ClaimsNotJson,
+                        MsalErrorMessage.ClaimsNotJson(claims),
+                        ex);
+                }
+
+                capabilitiesJson.Merge(claimsJson, new JsonMergeSettings
```

Comment:
I assume you'll always get a value there for claimsJson that can be merged

Reply:
Good point, I think so. I already check if claims is in JSON format. Capabilities are created by MSAL, in JSON format and I think 2 json objects can always merge.